### PR TITLE
Load default language if employee id lang does not exist

### LINKED ISSUE #30091: BO - An exception is displayed in all migrated page after deleting the employee language
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

In the BO > All migrated pages, An exception is displayed after deleting the employee's language.
I reproduce the same issue with @HanaRebaiQA & @MhiriFaten 

### Expected behavior

All pages are well displayed without exception.
In case of deletion of the language of an employee, this employee's language fallback to the default language of the shop. i.e : French is set as employee language and English is the default shop language, if we delete French from the shop, English is selected for the employee.

### Steps to reproduce

1. Install 8.0.0 with multilanguage (eg. English: default language & French)
2. Go to BO > Advanced Parameters > Team > Employees page
3. Create an employee with the French language
4. Go to BO > International > Languages page
5. Delete the French language
6. Try to Sign in with the new employee created
7. Try to visit any migrated page
8. See exception

image

### PrestaShop version(s) where the bug happened

8.0.0-rc1, 8.0.0-rc branch
It is ok with PS1.7.8.7, PS1.7.7.8

### PHP version(s) where the bug happened

7.2

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by mflasquin**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.0.x
| Description?      | Load default language if employee id lang does not exist
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #30091
| How to test?      | See #30091

### CODE REVIEW

**Review by 0x346e3730 — DISMISSED** [review]

File: src/PrestaShopBundle/EventListener/UserLocaleListener.php [hunk 1]
```diff
@@ -50,11 +66,18 @@ public function onKernelRequest(GetResponseEvent $event)
...
+    /**
+     * @return string
+     */
+    private function getLocaleFromEmployee(): string
     {
         $employee = $this->prestaShopContext->employee;
         $employeeLanguage = new Language($employee->id_lang);
 
+        if (!$employeeLanguage->locale) {
+            $employeeLanguage = new Language($this->configuration->get('PS_LANG_DEFAULT'));
```
Comment (FabienPapet) [inline src/PrestaShopBundle/EventListener/UserLocaleListener.php]: You should inject `prestashop.core.admin.lang.repository` to do this call

File: src/PrestaShopBundle/EventListener/UserLocaleListener.php [hunk 2]
```diff
@@ -26,20 +26,36 @@
 
 namespace PrestaShopBundle\EventListener;
 
+use Context;
```
Comment (FabienPapet) [inline src/PrestaShopBundle/EventListener/UserLocaleListener.php:29]: Could not find any usage of this class inside the class, where is it used ?

**Review by FabienPapet — CHANGES_REQUESTED** [review]

File: src/PrestaShopBundle/EventListener/UserLocaleListener.php (hunk 1 above)
Comment (FabienPapet) [reply src/PrestaShopBundle/EventListener/UserLocaleListener.php]: + when is  the `$employeeLanguage` variable used after getting the language ?

File: src/PrestaShopBundle/EventListener/UserLocaleListener.php (hunk 2 above)
Comment (mflasquin) [reply src/PrestaShopBundle/EventListener/UserLocaleListener.php:29]: In the phpdoc of $prestaShopContext

**Review by FabienPapet — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Review by AureRita — APPROVED** [review]
Hello @mflasquin 

Thank you for your PR, I tested it and it works ! 

Thank you

**Pull request merged**